### ISSUE TITLE
How to freeze layers that located at the last of the whole graph  when fine-tune

### ISSUE THREAD

**Issue by ztwe** [tensorflow/tensorflow#23684]
The grpah is A->B->C->loss, and I want to freeze A and C when fine-tune. A is a pretrained network. It's easy to freeze A. B contains LSTM and fc ops. C is variables or tuple tensors which are the results of fc. C stores one tensor  for one time because the tensor is hidden state of LSTM. Actually C will stores 6 tensors for a iteration.
It seems there are several solutions:
1. C is variables and "trainable=False"           (this results in "no gradient" error)
2. tf.stop_gradient()          (not try yet)
3.  add var_list to function optimize_loss()         (not try yet)

Which one should I try? Or are there new solutions?

**Comment by harshini-gadige** [comment]
This question is better asked on [StackOverflow](http://stackoverflow.com/questions/tagged//tensorflow) since it is not a bug or feature request. There is also a larger community that reads questions there.
